### PR TITLE
"polish activation"

### PR DESCRIPTION
**Pull request by dzhwinter**
fix https://github.com/PaddlePaddle/Paddle/issues/9416

### CODE REVIEW

File: paddle/fluid/operators/activation_op.h [hunk 1]
```diff
@@ -77,7 +77,16 @@ class ActivationGradKernel
...
     }
-    functor(*place, x, out, dout, dx);
+    bool inplace = functor.Inplace();
+    if (!inplace) {
+      auto* X = context.Input<framework::Tensor>("X");
+      auto x = framework::EigenVector<T>::Flatten(*X);
+      functor(*place, x, out, dout, dx);
+    } else {
+      framework::Tensor X;
+      auto x = framework::EigenVector<T>::Flatten(X);
```
Comment (panyx0718) [inline paddle/fluid/operators/activation_op.h]: can this flatten been avoided?

File: paddle/fluid/operators/activation_op.h [hunk 2]
```diff
@@ -101,6 +118,7 @@ struct SigmoidFunctor : public BaseActivationFunctor<T> {
 
 template <typename T>
 struct SigmoidGradFunctor : public BaseActivationFunctor<T> {
+  bool Inplace() const { return true; }
```
Comment (panyx0718) [inline paddle/fluid/operators/activation_op.h]: Can we have this in C++
static char* inplace_acts = ["sigmoid", "exp", ...]
bool IsInPlaceAct(const string& n) {
  return n in inplace_acts;
}
Inplace() { IsInPlaceAct("exp") }

in Python, we expose IsInPlaceAct() as core.is_in_place_act()

File: paddle/fluid/pybind/pybind.cc [hunk 3]
```diff
@@ -511,6 +512,9 @@ All parameter, weight, gradient are variables in Paddle.
         self.back().set_lod(t.lod());
       });
 
+  m.def("IsInplace",
```
[the remark this answers is not in the document]
Comment (dzhwinter) [reply paddle/fluid/pybind/pybind.cc]: This function bind c++ and Python, I thought that we should follow the c++ style where it been defined.

File: paddle/fluid/operators/CMakeLists.txt [hunk 4]
```diff
@@ -163,7 +163,12 @@ function(op_library TARGET)
 
     # pybind USE_OP
     if (${pybind_flag} EQUAL 0)
+      # NOTE(*): activation use macro to regist the kernels, set use_op manually.
+      if(${TARGET} STREQUAL "activation")
```
[the remark this answers is not in the document]
Comment (dzhwinter) [reply paddle/fluid/operators/CMakeLists.txt]: Yep. The CMakeLists in operators need a clean up, will file a PR do this later.

**Pull request merged**